GCP/DirectoryInsights/README.md

```diff
@@ -12,20 +12,78 @@ _Note: This document assumes the use of Python 3.9_
 
 ## Pre-requisites
 - [Your JumpCloud API key](https://docs.jumpcloud.com/2.0/authentication-and-authorization/authentication-and-authorization-overview)
+- Google Cloud Admin/Owner account with these roles:
+  - ```roles/serviceusage.serviceUsageAdmin```
+  - ```roles/cloudbuild.builds.editor```
+  - ```roles/resourcemanager.projects.setIamPolicy```
 - [GCLOUD CLI installed](https://cloud.google.com/sdk/docs/install)
-- [Google Cloud Build](https://cloud.google.com/build/docs/securing-builds/configure-access-for-cloud-build-service-account)
-  - Permissions:
-    - Cloud Functions = Enabled
-    - Cloud Scheduler = Enabled
-    - Secret Manager = Enabled
-  - Cloud Build Service account email: admin access to services above
-- JumpCloud administrator will need `Cloud Build Editor` role with their GCP account
-- Cloud Functions Invoker service account
+  - After installing the CLI, run ```gcloud auth login ``` and login with your Admin/Owner account
+- On your CLI, run these commands to enable the [services](https://cloud.google.com/apis?hl=en) needed to build the app:
+  - ```gcloud services enable cloudbuild.googleapis.com```
+  - ```gcloud services enable cloudfunctions.googleapis.com```
+  - ```gcloud services enable cloudscheduler.googleapis.com```
+  - ```gcloud services enable storage-component.googleapis.com```
+  - ```gcloud services enable secretmanager.googleapis.com```
+  - ```gcloud services enable cloudresourcemanager.googleapis.com```
+- Your GCP Project ID and Number, on your CLI you can run ```gcloud projects list``` to get your project's name and ID
+  - On your CLI create a variable for Project ID and Number so that you won't need to insert these variables on the commands below:
+    - ```PROJECTID=INSERTYOURGCPID```
+    - ```PROJECTNUM=INSERTYOURGCPNUM```
+- You must assign the Cloud Build Service account [roles](https://console.cloud.google.com/cloud-build/settings/). This account will have the suffix ```ProjectNumber@cloudbuild.gserviceaccount.com``` These can be removed after the build has succeeded:
+  - Cloud Functions Developer
+    - ```gcloud projects add-iam-policy-binding $PROJECTID --member=serviceAccount:$PROJECTNUM@cloudbuild.gserviceaccount.com --role=roles/cloudfunctions.developer```
+  - Service Account User
+    - ```gcloud projects add-iam-policy-binding sa-slackbot --member=serviceAccount:$PROJECTNUM@cloudbuild.gserviceaccount.com --role=roles/iam.serviceAccountUser```
+  - Secrets Manager Admin
+    - ```gcloud projects add-iam-policy-binding sa-slackbot --member=serviceAccount:$PROJECTNUM@cloudbuild.gserviceaccount.com --role roles/secretmanager.admin```
+  - Storage Admin
+    - ```gcloud projects add-iam-policy-binding sa-slackbot --member=serviceAccount:$PROJECTNUM@cloudbuild.gserviceaccount.com --role roles/storage.admin```
```

Comment:
This doesn't work 
![Screenshot 2024-07-24 at 11 44 48 AM](https://github.com/user-attachments/assets/9f37cfc9-c7c5-463c-90cb-848783ced68b)

Reply:
I stumbled upon this issue. Seems like there was a hidden character in the spacing between` .com` and `--role`. Kinda weird, it might have something to do with Notes autocorrection stuff which I have issue sometimes. I fixed the issue by deleting that space and adding it back. I know... odd